### PR TITLE
configuration connector component

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new `app-configuration-connector` component for enhanced onboarding experience.
  - Added UI elements for managing connections with accounting applications.

- **Enhancements**
  - Improved user interface with additional configuration options for organizational and accounting app details.

- **Style**
  - Implemented new styling classes for better layout control.

- **Documentation**
  - Updated module declarations to include the new `ConfigurationConnectorComponent`.

- **Refactor**
  - Removed unnecessary `div` elements to streamline the component structure.

- **Bug Fixes**
  - Adjusted a string parameter in the onboarding process for consistency.

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
different states of connectivity

Screenshot 2023-12-01 at 4 28 48 PM
 
Screenshot 2023-12-01 at 4 27 25 PM

Screenshot 2023-12-01 at 4 28 07 PM

Screenshot 2023-12-01 at 4 26 53 PM

**Comment by ashwin1111** [comment]
@coderabbitai review

**Comment by ashwin1111** [comment]
@coderabbitai review

File: src/app/integrations/business-central/business-central-onboarding/business-central-onboarding-connector/business-central-onboarding-connector.component.html [hunk 1]
```diff
@@ -13,9 +13,16 @@
...
+            <div>
+                <app-configuration-connector
+                [accountingAppTitle]="'Dynamics 365 Business Central'"
+                [accountingCompanyName]="'Dynamics 365 Business Central'"
+                [fyleOrgName]="'Fyle Testing Org'"
+                [accountingCompanyTokenExpired]="false"
+                [isAccountingCompanyConnected]="false"
+                [showDisconnect]="false"
+                [accountingCompanyConnectionInProgress]="false"
+                [switchLinkTex]="'Switch Entity'"></app-configuration-connector>
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/integrations/business-central/business-central-onboarding/business-central-onboarding-connector/business-central-onboarding-connector.component.html:25]: +1

File: src/app/shared/components/configuration/configuration-connector/configuration-connector.component.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                Please hold on; we are fetching data from your {{ accountingAppTitle }} account, which may take a few seconds.
+            </h4>
+            <h4 *ngIf="accountingCompanyTokenExpired && !accountingCompanyConnectionInProgress" class="tw-text-faded-text-color ">
+                Token expired
+            </h4>
+            <h4 *ngIf="!accountingCompanyConnectionInProgress && accountingCompanyTokenExpired" (click)="connect()" class="tw-text-mandatory-field-color">
+                Re-Connect to {{ accountingAppTitle }}
+            </h4>
+            <h4 *ngIf="!accountingCompanyConnectionInProgress && showDisconnect" (click)="disconnect()" class="tw-text-mandatory-field-color">
+                {{ switchLinkTex }}
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/shared/components/configuration/configuration-connector/configuration-connector.component.html:46]: +1

File: src/app/shared/components/configuration/configuration-connector/configuration-connector.component.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+})
+export class ConfigurationConnectorComponent implements OnInit {
+
+  @Input() fyleOrgName: string;
+
+  @Input() accountingAppTitle: string;
+
+  @Input() accountingCompanyName: string;
+
+  @Input() switchLinkTex: string;
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply src/app/shared/components/configuration/configuration-connector/configuration-connector.component.ts:16]: +1

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**